react/components/Logo.js

```diff
@@ -1,67 +1,81 @@
-import ReactResizeDetector from 'react-resize-detector'
+import { string, bool, shape } from 'prop-types'
+import { ExtensionPoint, Link, withRuntimeContext } from 'vtex.render-runtime'
+import { CONSTANTS } from './Helpers'
 
 import header from '../store-header.css'
 
-const Logo = ({ link, src, title, sizeMobile, sizeDesktop, onResize }) => (
-  <div className={`${header.topMenuLogo} pv2`}>
-    <ReactResizeDetector handleHeight onResize={onResize}>
-      <Link
-        to={link}
-        // there is a weird bottom padding being added
-        // below the image. This fixes the issue,
-        // but the cause should be investigated
-        className={`outline-0 ${header.logoLink}`}
-      >
-        {sizeMobile && (
-          <div className="db dn-ns">
-            <ExtensionPoint
-              id="logo"
-              url={src}
-              title={title}
-              width={sizeMobile.width}
-              height={sizeMobile.height}
-              isMobile={true}
-            />
-          </div>
-        )}
-        {sizeDesktop && (
-          <div className="dn db-ns">
-            <ExtensionPoint
-              id="logo"
-              url={src}
-              title={title}
-              width={sizeDesktop.width}
-              height={sizeDesktop.height}
-              isMobile={false}
-            />
-          </div>
-        )}
-      </Link>
-    </ReactResizeDetector>
+const Logo = ({
+  link,
+  src,
+  title,
+  size,
+  runtime: { hints: { mobile, desktop } }
+}) => (
+  <div className={`${header.topMenuLogo} pv2 mr5`}>
+    <Link
+      to={link}
+      className={`outline-0 ${header.logoLink}`}
+    >
+      { mobile && (
+        <div className="db dn-ns">
+          <ExtensionPoint
+            id="logo"
+            url={src}
+            title={title}
+            width={size.mobile.width}
+            height={size.mobile.height}
+            isMobile={true}
+          />
+        </div>
+      )}
+
+      { desktop && (
+        <div className="dn db-ns">
+          <ExtensionPoint
+            id="logo"
+            url={src}
+            title={title}
+            width={size.desktop.width}
+            height={size.desktop.height}
+            isMobile={false}
+          />
+        </div>
+      )}
+    </Link>
   </div>
 )
 
 Logo.propTypes = {
-  src: PropTypes.string.isRequired,
-  link: PropTypes.string,
-  title: PropTypes.string,
-  onResize: PropTypes.func,
-  sizeMobile: PropTypes.shape({
-    width: PropTypes.number.isRequired,
-    height: PropTypes.number.isRequired,
-  }),
-  sizeDesktop: PropTypes.shape({
-    width: PropTypes.number.isRequired,
-    height: PropTypes.number.isRequired,
-  }),
+  src: string.isRequired,
+  link: string,
+  title: string,
+  mobile: bool,
+  desktop: bool,
+  size: shape({
+    desktop: shape({
+      width: string,
+      height: string
+    }),
+    mobile: shape({
+      width: string,
+      height: string
+    })
```

Comment:
Add `runtime` here... And also, there are some non-required props not describe on defaults

Reply:
✅